### PR TITLE
New Resource: Web Application Firewall Policy

### PR DESCRIPTION
**Pull request by jeffreyCline**

### CODE REVIEW

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+* `resource_group_name` - (Required) The name of the resource group. Changing this forces a new resource to be created.
+
+* `location` - (Optional) Resource location. Changing this forces a new resource to be created.
+
+* `custom_rules` - (Optional) One or more `custom_rule` block defined below.
+
+* `policy_settings` - (Optional) One `policy_setting` block defined below.
+
+* `tags` - (Optional) Resource tags. Changing this forces a new resource to be created.
```
Comment (nexxai) [inline website/docs/r/web_application_firewall_policy.html.markdown]: Changing a tag results in a new policy being created?

Comment (jeffreyCline) [reply website/docs/r/web_application_firewall_policy.html.markdown]: @nexxai No... that was my bad. Thanks for nice catch. 😆

File: azurerm/resource_arm_web_application_firewall_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.match_variables.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.match_variables.0.variable_name", "RequestHeaders"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.match_variables.0.selector", "UserAgent"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.operator", "Contains"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.negation_condition", "false"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.match_values.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.match_conditions.1.match_values.0", "Windows"),
+					resource.TestCheckResourceAttr(resourceName, "custom_rules.1.action", "Block"),
+				),
+			},
```
Comment (katbyte) [inline azurerm/resource_arm_web_application_firewall_policy_test.go]: Could we add an import check step here?

File: azurerm/provider.go [hunk 3]
```diff
@@ -327,8 +327,14 @@ func Provider() terraform.ResourceProvider {
 		"azurerm_private_dns_cname_record":                                               resourceArmPrivateDnsCNameRecord(),
 		"azurerm_public_ip":                                                              resourceArmPublicIp(),
 		"azurerm_public_ip_prefix":                                                       resourceArmPublicIpPrefix(),
+		"azurerm_recovery_network_mapping":                                               resourceArmRecoveryServicesNetworkMapping(),
```
Comment (katbyte) [inline azurerm/provider.go]: I think you need to rebase/merge?

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 4]
```diff
@@ -0,0 +1,145 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_web_application_firewall_policy"
+sidebar_current: "docs-azurerm-resource-web-application-firewall-policy"
+description: |-
+  Manage a Azure Web Application Firewall Policy instance.
```
Comment (katbyte) [inline website/docs/r/web_application_firewall_policy.html.markdown]:
```suggestion
  Manages a Azure Web Application Firewall Policy instance.
```

File: website/docs/r/web_application_firewall_policy.html.markdown [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+---
+
+# azurerm_web_application_firewall_policy
+
+Manage a Azure Web Application Firewall Policy instance.
```
Comment (katbyte) [inline website/docs/r/web_application_firewall_policy.html.markdown]:
```suggestion
Manages a Azure Web Application Firewall Policy instance.
```

File: azurerm/resource_arm_web_application_firewall_policy_test.go (hunk 2 above)
Comment (jeffreyCline) [reply azurerm/resource_arm_web_application_firewall_policy_test.go]: Yep.

File: azurerm/provider.go (hunk 3 above)
Comment (jeffreyCline) [reply azurerm/provider.go]: I think this is right... I put them in correct alphabetical order.

Comment (katbyte) [reply azurerm/provider.go]: Except these were added in a different PR not this one?

**Comment by jeffreyCline** [comment]
image

**Pull request merged**